backend/de.metas.banking/de.metas.banking.base/src/main/java/de/metas/banking/importfile/BankStatementImportFileRepository.java

```diff
@@ -0,0 +1,75 @@
+/*
+ * #%L
+ * de.metas.banking.base
+ * %%
+ * Copyright (C) 2022 metas GmbH
+ * %%
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as
+ * published by the Free Software Foundation, either version 2 of the
+ * License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public
+ * License along with this program. If not, see
+ * <http://www.gnu.org/licenses/gpl-2.0.html>.
+ * #L%
+ */
+
+package de.metas.banking.importfile;
+
+import lombok.NonNull;
+import org.adempiere.banking.model.I_C_BankStatement_Import_File;
+import org.adempiere.exceptions.AdempiereException;
+import org.adempiere.model.InterfaceWrapperHelper;
+import org.springframework.stereotype.Repository;
+
+import static org.adempiere.model.InterfaceWrapperHelper.saveRecord;
+
+@Repository
+public class BankStatementImportFileRepository
+{
+	@NonNull
+	public BankStatementImportFile getById(@NonNull final BankStatementImportFileId bankStatementImportFileId)
+	{
+		return ofRecord(getRecordById(bankStatementImportFileId));
+	}
+
+	public void processRecord(@NonNull final BankStatementImportFileId bankStatementImportFileId)
```

Comment:
i don't like repo-methods such as processRecord(@NonNull final BankStatementImportFileId bankStatementImportFileId)

in my mind, the repo should be used to load and store only.
and then the user can load a pojo (domain object) where the processed field has a setter.
the can then set it to true and call the repo's save method with the altered domain object.

(posted the same question/comment to @teosarca via DM)